src/main/java/gg/projecteden/nexus/features/shops/providers/ResourceWorldMarketProvider.java

```diff
@@ -0,0 +1,67 @@
+package gg.projecteden.nexus.features.shops.providers;
+
+import fr.minuskube.inv.ClickableItem;
+import fr.minuskube.inv.content.InventoryContents;
+import gg.projecteden.nexus.features.shops.providers.common.ShopProvider;
+import gg.projecteden.nexus.models.shop.Shop;
+import gg.projecteden.nexus.models.shop.Shop.Product;
+import gg.projecteden.nexus.models.shop.ShopService;
+import gg.projecteden.nexus.utils.ItemBuilder;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ResourceWorldMarketProvider extends ShopProvider {
+	private static final ShopService service = new ShopService();
+	private static final Shop market = service.getMarket();
+
+	public ResourceWorldMarketProvider(ShopProvider previousMenu) {
+		this.previousMenu = previousMenu;
+	}
+
+	@Override
+	public void open(Player player, int page) {
+		open(player, page, this, "Resource World Market");
+	}
+
+	@Override
+	public void init(Player player, InventoryContents contents) {
+		super.init(player, contents);
+
+		final Shop shop = service.get(player);
+		final List<Product> products = market.getProducts().stream().filter(Product::isResourceWorld).toList();
```

Comment:
```suggestion
		final List<Product> products = ResourceWorldCommand.getResourceWorldProducts();
```
(needs to be imported)